src/core/src/local_loggers/StdOutFileMirror.py

```diff
@@ -34,7 +34,11 @@ def __init__(self, env_layer, file_logger):
             sys.stdout.write("WARNING: StdOutFileMirror - Skipping as FileLogger is not initialized")
 
     def write(self, message):
-        self.terminal.write(message)  # enable standard job output
+        try:
+            self.terminal.write(message)  # enable standard job output
+        except Exception as error:
+            # Ignore any exceptions from writing to the terminal which is mostly observed to be captured by the dead extension handler process.
```

Comment:
Use a flag within the class to log to file the first time an error is seen (repeated errors can be ignored as such).

This allows us to be sure this has actually caught the issue, and also help us keep tabs on the health of this overall.

Reply:
Please simulate this failure (after addressing the first comment about additional logging) and validate -- insert an long-running loop outputting a lot of text for a long duration of time early in Core's execution. Invoke everything in test as you normally would and let it run in the background. Please review the logs for the expected outcome and to see the mitigation work, and confirm here please. 

It's possible to be know with little overhead if this addresses the full extent of the issue before deploying, and is cheaper overall.

Reply:
Tested this locally with a long running loop at the beginning of PatchInstaller. 

- The core was triggered by the first method (where errors are sent to devnull) and handler goes away. 

- Core stops when it's trying to print a long text and it doesn't go into the exception handling part.

- Also, tried everything with using subprocess.PIPE for error logging while launching core the first time. The process after the exeception is not consistent. It intermittently goes to the exception handling part and completes the remaining operations but there are instances when it either stops even before exception handler (i.e. does not write the error in core log file) or it writes the error to the log file and then stops, without completing the remaining steps.

So in both the cases (logging errors in devnull or subprocess.PIPE), core usually stops executing after the error. I'm thinking of limiting the # of characters being written in one print statement. If the text exceeds this limit, we can split the message into multiple lines. Let me know what you think

Reply:
Comment out self.terminal.write and keep the test running again - let me know if the forced repro's behavior changes in any way. The outcome of this can inform what needs to happen next.